src/io/flutter/run/daemon/DevToolsService.java

```diff
@@ -45,6 +53,19 @@ public static DevToolsService getInstance(@NotNull final Project project) {
 
   private DevToolsService(@NotNull final Project project) {
     this.project = project;
+
```

Comment:
doesn't this file need an API to query to make sure devtools is done serving? Otherwise we'll launch devtools URLs that fail? a user could run an app very quickly after opening the project and hit a case where devtools isn't fully running yet.

Reply:
Yes, I was thinking a method like `getDevToolsUrl` would return a future but wasn't planning to add in this change. Is that what you mean?

Reply:
Yes. That is what I mean.